src/DataStreams/RemoteQueryExecutor.cpp

```diff
@@ -245,6 +288,19 @@ Block RemoteQueryExecutor::read()
     }
 }
 
+void RemoteQueryExecutor::setUuids(const std::vector<UUID> & uuids)
+{
+    Context & query_context = const_cast<Context &>(context).getQueryContext();
+    auto duplicates = query_context.getUUIDs()->add(uuids);
+
+    if (!duplicates.empty())
+    {
+        std::lock_guard lock(duplicated_uuids_mutex);
+        duplicated_uuids.insert(duplicated_uuids.begin(), duplicates.begin(), duplicates.end());
+        throw Exception("Found duplicate uuids while processing query.", ErrorCodes::DUPLICATED_UUIDS);
```

Comment:
Why not return a bool here instead of throwing exceptions? Also, `[[nodiscard]] appendProcessedPartUuuids` seems better to me.